alacritty.yml

```diff
@@ -486,26 +474,47 @@ debug:
 #   - ClearHistory
 #   - Hide
 #   - Quit
-#   - ClearLogNotice
-#   - SpawnNewInstance
 #   - ToggleFullscreen
+#   - SpawnNewInstance
+#   - ClearLogNotice
 #   - None
 #
-# Values for `action` (macOS only):
+#   (macOS only):
 #   - ToggleSimpleFullscreen: Enters fullscreen without occupying another space
 #
-# Values for `command`:
-#   The `command` field must be a map containing a `program` string and
-#   an `args` array of command line parameter strings.
+# - `command`: Fork and execute a specified command plus arguments
+#
+#    The `command` field must be a map containing a `program` string and an
+#    `args` array of command line parameter strings. For example:
+#       `{ program: "alacritty", args: ["-e", "vttest"] }`
+#
+# And optionally:
+#
+# - `mods`: Terminal modes where the binding is applicable
+#
+#    - Command
+#    - Control
+#    - Option
+#    - Super
+#    - Shift
+#    - Alt
+#
+#    Multiple `mods` can be combined using `|` like this:
+#       `mods: Control|Shift`.
+#    Whitespace and capitalization are relevant and must match the example.
+#
+# - `mode`: ...
```

Comment:
I'm struggling a bit to fill in these last `...`s. I can see the complete list (embeded below), but only `ALT_SCREEN`, `APP_KEYPAD`, and `APP_CURSOR` are exposed through the config.

https://github.com/jwilm/alacritty/blob/55d346c7d44215c51187cafcac79e3504209f942/alacritty_terminal/src/term/mod.rs#L489-L508

However for the config values we only need to document `AppCursor`, `AppKeypad`, and `Alt`.

Reply:
> However for the config values we only need to document AppCursor, AppKeypad, and Alt.

That's correct, only the values relevant for keybindings are available, so only those need to be documented. That shouldn't be a problem, right?

Reply:
Ah, if I'm understanding this correctly, we just reexport the mode through `AppCursor`, since `App` is appropriate for a Digital Equipment Corporation private control sequence :trollface:. And there are differences in the escape sequences between modes. I'd never dug into this before. How strange!

It would be really nice if we could abstract this away from the user config altogether. But that's another story.


```
ansi::Mode::CursorKeys => self.mode.insert(TermMode::APP_CURSOR),
```

> DECCKM – Cursor Keys Mode (DEC Private)
>
> This is a private parameter applicable to set mode (SM) and reset mode (RM) control sequences. This mode is only effective when the terminal is in keypad application mode (see DECKPAM) and the ANSI/VT52 mode (DECANM) is set (see DECANM). Under these conditions, if the cursor key mode is reset, the four cursor function keys will send ANSI cursor control commands. If cursor key mode is set, the four cursor function keys will send application functions.

Reply:
If we abstract this away from user config, the issue is that the user doesn't have full control about the special bindings anymore.

Sure we could set all our default bindings to never get overwritten at all, but I don't think that would actually benefit the user at all.

And the app keypad/app cursor modes are actually quite common.